### PR TITLE
add safe execute in action

### PR DESCRIPTION
**Pull request by vincenzosarnataro**

### CODE REVIEW

**Review by sma6871 — COMMENTED** [review]
Why do we need this safe execute?

**Comment by fabiosassu-sysdata** [comment]
> Why do we need this safe execute?

Hi sma6871,
The "safeExecute" method call is meant to be used if you plan a different behaviour based on the error of your response.
If you use the simple "execute" call, whenever there's an error, the method will throw an exception and you will have to handle it by try-catching it and defining a behaviour for that usecase.
If you use the "safeExecute" method call, whenever there's an error, it will fallback to an "InternalError" failure in case of an exception, saving you from an application crash.
Moreover, if you'd like to customize the behaviour in a general way, you can define a class that extends from the SafeExecuteInterface and inside the overridden safeExecute call you can define the custom behaviour you want for your application.
Finally,  you'll have to assign the custom SafeExecutor as the BaseConfig.safeEcecutor variable.
It is suggested that you assign this variable as soon as possible (i.e. in the Application's class onCreate method).

File: ktandroidarchitecturecore/src/main/java/it/sysdata/ktandroidarchitecturecore/platform/SafeExecute.kt [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package it.sysdata.ktandroidarchitecturecore.platform
+
+import it.sysdata.ktandroidarchitecturecore.exception.Failure
+import it.sysdata.ktandroidarchitecturecore.functional.Either
+import kotlinx.coroutines.Deferred
+
+class SafeExecute : SafeExecuteInterface {
```
Comment (aguitto) [inline ktandroidarchitecturecore/src/main/java/it/sysdata/ktandroidarchitecturecore/platform/SafeExecute.kt:12]: Comment this class

File: ktandroidarchitecturecore/src/main/java/it/sysdata/ktandroidarchitecturecore/platform/SafeExecuteInterface.kt [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package it.sysdata.ktandroidarchitecturecore.platform
+
+import it.sysdata.ktandroidarchitecturecore.exception.Failure
+import it.sysdata.ktandroidarchitecturecore.functional.Either
+import kotlinx.coroutines.Deferred
+
+interface SafeExecuteInterface {
```
Comment (aguitto) [inline ktandroidarchitecturecore/src/main/java/it/sysdata/ktandroidarchitecturecore/platform/SafeExecuteInterface.kt:11]: Comment this interface

File: ktandroidarchitecturecore/src/main/java/it/sysdata/ktandroidarchitecturecore/interactor/Action.kt [hunk 3]
```diff
@@ -51,7 +51,17 @@ class Action<Params : ActionParams, Model : Any, UiModel : Any> private construc
         loadingLiveData.value = true
         uc.execute({ it.either(::handleFailure, ::handleSuccess) }, params, scope)
     }
+    /**
+     * Execute the action in safe mode
```
Comment (aguitto) [inline ktandroidarchitecturecore/src/main/java/it/sysdata/ktandroidarchitecturecore/interactor/Action.kt]: add more details

**Review by sma6871 — APPROVED** [review]

**Pull request merged**